src/input.cc

```diff
@@ -605,25 +628,60 @@ void screenshotHandlerConfigure(int keyCode, ScreenshotHandler* handler)
     gScreenshotHandler = handler;
 }
 
+// Unused, this function changed meaning: it is now the patched GetTickCount() from sfall SpeedPatch, while the non-patched ones have become SDL_GetTicks().
+// Note that other files still _do_ use getTicksSince() which _does_ return the multiplied (sped up) ticks.
 // 0x4C9370
-unsigned int getTicks()
+unsigned int getMultipliedTicks()
+{
+    unsigned int newTickCount = SDL_GetTicks();
+    if (newTickCount == gLastTickCount) return gStoredTickCount;
+
+    // Just in case someone's been running their computer for 49 days straight
+    if (gLastTickCount > newTickCount) {
+        gLastTickCount = newTickCount;
+        return gStoredTickCount;
+    }
+
+    float elapsed = static_cast<float>(newTickCount - gLastTickCount);
+    gLastTickCount = newTickCount;
+
+    // Multiply the tick count difference by the multiplier
+    /* TODO janiczek: Original condition was:
+       if (IsGameLoaded() && enabled && (!(mode = GetLoopFlags()) || mode == LoopFlag::COMBAT || mode == (LoopFlag::COMBAT | LoopFlag::PCOMBAT) || (mode & LoopFlag::WORLDMAP)) && !slideShow)
+    */
+    if (gSpeedPatchEnabled) {
+        elapsed *= gSpeedMulti;
+        elapsed += gTickCountFraction;
+        gTickCountFraction = modff(gTickCountFraction, &gTickCountFraction);
+    }
+    
+    gStoredTickCount += static_cast<unsigned int>(elapsed);
+
+    return gStoredTickCount;
+}
+
+// Done by sfall (presumably) because this time gets converted to game time.
+// If you've played the game at 8x speed for a while, you'll want the time in
+// the savefile to agree with the time in the game at the time you saved it.
+time_t getLocalTimeAfterSpeedup()
 {
-    return SDL_GetTicks();
+    return gStartTime + gStoredTickCount / 1000;
 }
 
 // 0x4C937C
 void inputPauseForTocks(unsigned int delay)
 {
+    // TODO janiczek: which one of these is 0x4C9384? Only one needs to be patched.
     // NOTE: Uninline.
-    unsigned int start = getTicks();
-    unsigned int end = getTicks();
+    unsigned int start = SDL_GetTicks();
+    unsigned int end = SDL_GetTicks();
```

Comment:
Do you have any pointers on how to figure out which C++ line corresponds to a given offset from sfall? Do I need to disassemble original fallout2.exe?

Reply:
You don't have to do much disassembling yourself, but you need IDA to check what's at the offsets. There's a database for it linked in Sfall's readme.

Reply:
Thank you, IDA Freeware was enough for me to make sense of the assembly and which call it was. (It was the first one.)